### ISSUE TITLE
Note length bug

### ISSUE THREAD

**Issue by rualark** [paulrosen/abcjs#570]
```
%%barnumbers 1
M:6/4
K:C
L:1/16
V: V0 clef=treble name="Sop."
[V: V0]B24|c16c4z4|d1c1c1c1z5d1c1c1c1z1z1z1z1z1z1z1z1z1c2-|c14z2z8|
```

Results in 

![image](https://user-images.githubusercontent.com/15198808/101351687-01252100-38a2-11eb-9514-aeb00c90cb18.png)

https://artinfuser.com/harmony-dev/editor.html?state=IgYAwgggzAAAgDGANhyBgAQOAoNAXAJwFMAjAGyJgEoAMRGAKgAR6GN6Z3EBqVRvhgL4wAoPV6shkjsNkz59OYoUAgBitYA4DhvoAIRHAwBlAPYAHAHQBMAAoFTJAIbkiAAhIBXPHlMA7N08.AEs8AE8AWgAOAFYAJjsHZ1cIrx9.CKDQyNi4gBiOegA0AH0AOYBCAF2gA__

**Comment by rualark** [comment]
```
%%barnumbers 1
M:6/4
K:C
L:1/16
V: V0 clef=treble name="Sop."
[V: V0]B24|c16c4z4|d1c1c1c1z5d1c1c1c1z1z1z1z1z1z1z1z1c3-|c13z1z2z8|
```

Results in 1/16th dotted at the end of the bar instead of 1/8th dotted note:
![image](https://user-images.githubusercontent.com/15198808/101351979-7c86d280-38a2-11eb-8f05-5326da56ca3e.png)

https://artinfuser.com/harmony-dev/editor.html?state=IgYAwgggzAAAgDGANhyBgAQOAoNAXAJwFMAjAGyJgEoAMRGAKgAR6GN6Z3EBqVRvhgL4wAoPV6shkjsNkz59OYsQAgBiFYBYBao4AIRHAwBlAPYAHAHQBMAAoFTJAIbkiAAhIBXPHlMA7N08.AEs8AE8AWgAOAFYAJjsHZ1cIrx9.CKDQyNi4gBiOegBUAH0AOYAiAFqgA__

**Comment by paulrosen** [comment]
Here's a simple case:
```
L:1/16
K:C
dczc2
```
The problem is when there is a rest internal to the beam.

A work around for now is to put a space before the `c2` so it isn't beamed.

**Comment by p-himik** [comment]
I'm not sure if it's related but it seems like there's an issue with fractional note lengths as well:
```
L:1/4
K:C
A2/3
```
is rendered as a dotted 8th note.

**Comment by paulrosen** [comment]
ABC can express things that can't be written in standard notation - abcjs does its best and picks the closest legal note length. So it is treating it as `A3/4`

**Comment by p-himik** [comment]
Right. Do you think it would be possible to add an ability to issue a warning or call some callback or add any other way to notify the user that the visual representation is not quite correct?
To give some context - it comes from a real application where a musician had to spend some time figuring out what was wrong with the notation but was relying too much on the visual representation.

**Comment by paulrosen** [comment]
Maybe. I guess I can enumerate all the legal note values - perhaps down to 1/64 note with three dots. There probably a formula that would work, too.
